ospd_openvas/preferencehandler.py

```diff
@@ -0,0 +1,610 @@
+            for vt_param_id, vt_param_value in vt_params.items():
+                param_type = self.get_vt_param_type(
+                    vtid, vt_param_id, vts_cache
+                )
+                param_name = self.get_vt_param_name(
+                    vtid, vt_param_id, vts_cache
+                )
+
+                if not param_type or not param_name:
+                    logger.debug(
+                        'Missing type or name for VT parameter %s of %s. '
+                        'It could not be loaded.',
+                        vt_param_id,
+                        vtid,
+                    )
+                    continue
+
+                if vt_param_id == '0':
+                    type_aux = 'integer'
+                else:
+                    type_aux = param_type
+
+                if self.check_param_type(vt_param_value, type_aux):
+                    logger.debug(
+                        'The VT parameter %s for %s could not be loaded. '
+                        'Expected %s type for parameter value %s',
+                        vt_param_id,
+                        vtid,
+                        type_aux,
+                        str(vt_param_value),
+                    )
+                    continue
+
+                if type_aux == 'checkbox':
+                    vt_param_value = _from_bool_to_str(int(vt_param_value))
+
+                vts_params[
+                    "{0}:{1}:{2}:{3}".format(
+                        vtid, vt_param_id, param_type, param_name
+                    )
+                ] = str(vt_param_value)
+
+        return vts_list, vts_params
+
+    def set_plugins(self, vts_cache: Dict[str, Dict],) -> bool:
+        """ Get the plugin list to be launched from the Scan Collection
+        and prepare the vts preferences. Store the data in the kb.
+        """
+        nvts = self.scan_collection.get_vts(self.scan_id)
+        self.scan_collection.release_vts_list(self.scan_id)
+        if nvts:
+            nvts_list, nvts_params = self.process_vts(nvts, vts_cache)
+            # Add nvts list
+            separ = ';'
+            plugin_list = 'plugin_set|||%s' % separ.join(nvts_list)
+            self.kbdb.add_scan_preferences(self.openvas_scan_id, [plugin_list])
+
+            # Add nvts parameters
+            for key, val in nvts_params.items():
+                item = '%s|||%s' % (key, val)
+                self.kbdb.add_scan_preferences(self.openvas_scan_id, [item])
+
+            nvts_params = None
+            nvts_list = None
+            item = None
+            plugin_list = None
+            nvts = None
+
+            return True
+
+        return False
+
+    @staticmethod
+    def build_alive_test_opt_as_prefs(target_options: Dict) -> List[str]:
+        """ Parse the target options dictionary.
+        @param credentials: Dictionary with the target options.
+
+        @return A list with the target options in string format to be
+                added to the redis KB.
```

Comment:
We should always use the [google style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) for the docstrings.
```suggestion
        Arguments:
            credentials: Dictionary with the target options.

        Return:
            A list with the target options in string format to be
                added to the redis KB.
```